services/location.js

```diff
@@ -1,44 +1,71 @@
 import lowerCase from 'lodash/lowerCase';
 import startCase from 'lodash/startCase';
-import { cartoRequest } from 'utils/request';
+
 import { getGeodescriberByGeostore } from 'services/geodescriber';
 import { getDatasetQuery } from 'services/datasets';
 import { getArea } from 'services/areas';
+import {
+  getCountriesProvider,
+  getRegionsProvider,
+  getSubRegionsProvider,
+} from 'services/country';
 
-export const countryConfig = {
-  adm0: (params) =>
-    cartoRequest(
-      `/sql?q=SELECT iso, name_engli as name FROM gadm36_countries WHERE iso = '${params.adm0}' AND iso != 'XCA' AND iso != 'TWN'`
-    ).then((response) => {
-      const { name, ...props } = response?.data?.rows?.[0];
-
-      return {
-        locationName: name,
-        ...props,
-      };
-    }),
-  adm1: (params) =>
-    cartoRequest(
-      `/sql?q=SELECT iso, gid_1 as id, name_0 as adm0, name_1 as adm1 FROM gadm36_adm1 WHERE gid_1 = '${params.adm0}.${params.adm1}_1' AND iso != 'XCA' AND iso != 'TWN'`
-    ).then((response) => {
-      const { adm1, adm0, ...props } = response?.data?.rows?.[0];
-
-      return {
-        locationName: `${adm1}, ${adm0}`,
-        ...props,
-      };
-    }),
-  adm2: (params) =>
-    cartoRequest(
-      `/sql?q=SELECT gid_2, name_0 as adm0, name_1 as adm1, name_2 as adm2 FROM gadm36_adm2 WHERE gid_2 = '${params.adm0}.${params.adm1}.${params.adm2}_1' AND iso != 'XCA' AND iso != 'TWN'`
-    ).then((response) => {
-      const { adm2, adm1, adm0, ...props } = response?.data?.rows?.[0];
+const findByIso = (list, iso) => list?.find((item) => item?.iso === iso);
+const findById = (list, idPrefix) =>
+  list?.find((item) => item?.id.startsWith(idPrefix));
 
-      return {
-        locationName: `${adm2}, ${adm1}, ${adm0}`,
-        ...props,
-      };
-    }),
+export const countryConfig = {
```

Comment:
I would encourage you to try to remove the duplication in `countryConfig`.
`adm0`, `adm1`, and `adm2` build on each other. Your current implementation reimplements the previous `adm` level in each case.

Reply:
You are completely right. I was thinking about this last week. I've decided the effort to create a generic method for this configuration wouldn't be worth since the methods I'm reusing here doesn't fit the return I need. I'll try a different approach though.

Reply:
Love it, Will! Nice refactor! 💪